aspnetcore/fundamentals/servers/httpsys.md

```diff
@@ -0,0 +1,146 @@
+
+* Call the [`UseHttpSys`](http://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNetCore/Hosting/WebHostBuilderKestrelExtensions/index.html#Microsoft.AspNetCore.Hosting.WebHostBuilderHttpSysExtensions.UseHttpSys.md) extension method on [WebHostBuilder](http://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNetCore/Hosting/WebHostBuilder/index.html#Microsoft.AspNetCore.Hosting.WebHostBuilder.md) in your `Main` method, specifying any HttpSys [options](https://github.com/aspnet/HttpSysServer/blob/rel/2.0.0/src/Microsoft.AspNetCore.Server.HttpSys/HttpSysOptions.cs) that you need, as shown in the following example:
+
+  [!code-csharp[](HttpSys/sample/Program.cs?name=snippet_Main&highlight=13-18)]
+
+* Configure URLs and ports to listen on 
+
+  By default ASP.NET Core binds to `http://localhost:5000`. To configure URL prefixes and ports, you can use the `UseUrls` extension method, the `urls` command-line argument, or the `UrlPrefixes` property on [HttpSysOptions](https://github.com/aspnet/HttpSysServer/blob/rel/2.0.0/src/Microsoft.AspNetCore.Server.HttpSys/HttpSysOptions.cs). The preceding code example uses `UrlPrefixes`.
+
+  An advantage of `UrlPrefixes` is that you get an error message immediately if you try to add a prefix that is formatted wrong. An advantage of `UseUrls` is that you can more easily switch between Kestrel and HttpSys. (But this applies only if you don't use SSL, because you can't use SSL with `UseUrls` and Kestrel.)
+
+  If you use both `UseUrls` and `UrlPrefixes`, the settings in `UrlPrefixes` override the ones in `UseUrls`. For more information, see [Hosting](../../fundamentals/hosting.md).
+
+  HttpSys uses the [Http.Sys prefix string formats](https://msdn.microsoft.com/library/windows/desktop/aa364698.aspx). There are no prefix string format requirements that are specific to HttpSys.
+
+  > [!NOTE]
+  > Make sure that you specify the same prefix strings in `UseUrls` or `UrlPrefixes` that you preregister on the server. 
+
+* Make sure your application is not configured to run IIS or IIS Express.
+
+  In Visual Studio, the default launch profile is for IIS Express.  To run the project as a console application you have to manually change the selected profile, as shown in the following screen shot.
+
+  ![Select console app profile](HttpSys/_static/vs-choose-profile.png)
+
+## Preregister URL prefixes and configure SSL
+
+Both IIS and HttpSys rely on the underlying Http.Sys kernel mode driver to listen for requests and do initial processing. In IIS, the management UI gives you a relatively easy way to configure everything. However, if you're using HttpSys you need to configure Http.Sys yourself. The built-in tool for doing that is netsh.exe. 
+
+The most common tasks you need to use netsh.exe for are reserving URL prefixes and assigning SSL certificates.
+
+NetSh.exe is not an easy tool to use for beginners. The following example shows the bare minimum needed to reserve URL prefixes for ports 80 and 443:
```

Comment:
I don't really like `NetSh.exe is not an easy tool to use for beginners.` but can't come up with anything better. Maybe
NetSh.exe can be difficult to use. The following example shows the ~bare~ minimum needed to reserve URL prefixes for ports 80 and 443: